tests/test_widgets.py

```diff
@@ -1832,6 +1834,105 @@ def testExecution(self):
+        self.model_mock.return_value.instrument_controlled = TestSignal()
+
+        points = np.rec.array([([0.0, 1.0, 2.0], False), ([2.0, 0.0, 1.0], True), ([0.0, 1.0, 1.0], True)],
+                              dtype=POINT_DTYPE)
+        self.model_mock.return_value.fiducials = points
+        self.presenter = MainWindowPresenter(self.view)
+        self.view.scenes = mock.create_autospec(SceneManager)
+        self.view.presenter = self.presenter
+        self.model_mock.return_value.alignment = None
+        self.dialog = CurrentCoordinatesDialog(self.view)
+
+    @staticmethod
+    def createPositioner():
+        q1 = Link("X", [1.0, 0.0, 0.0], [0.0, 0.0, 0.0], Link.Type.Prismatic, -200.0, 200.0, 0)
+        return SerialManipulator("", [q1])
+
+    @staticmethod
+    def createPositioningStack():
+        y_axis = create_cuboid(200, 10, 200)
+        z_axis = create_cylinder(25, 50)
+        q1 = Link("Z", [0.0, 0.0, 1.0], [0.0, 0.0, 0.0], Link.Type.Revolute, -3.14, 3.14, 0, z_axis)
+        q2 = Link("Y", [0.0, 1.0, 0.0], [0.0, 0.0, 0.0], Link.Type.Prismatic, -200.0, 200.0, 0, y_axis)
+        s = SerialManipulator("", [q1, q2], custom_order=[1, 0], base=Matrix44.fromTranslation([0.0, 0.0, 50.0]))
+        return PositioningStack(s.name, s)
+
+    @mock.patch('sscanss.app.dialogs.misc.np.savetxt', autospec=True, return_value=True)
+    @mock.patch('sscanss.app.dialogs.misc.QtWidgets.QFileDialog.getSaveFileName', return_value=('dummy', None))
+    def testCurrentCoordinatesDialog(self, dialog_mock, savetxt):
+        savetxt.assert_not_called()
+        # Test non aligned case that fiducials are parsed unchanged
+        self.assertEqual(self.model_mock.return_value.fiducials['points'][1, 1],
+                         float(self.dialog.ftable_widget.item(1, 1).text()))
+        self.assertEqual(self.model_mock.return_value.fiducials['points'][2, 2],
+                         float(self.dialog.ftable_widget.item(2, 2).text()))
+
+        # Should not export fiducial coordinates if sample not aligned on instrument
+        self.dialog.export_fiducials_button.click()
+        dialog_mock.assert_not_called()
+        savetxt.assert_not_called()
+
+        # Align sample on instrument at zero and test fiducial is where expected
+        self.mock_instrument.positioning_stack.fkine([0, 0], set_point=True)
+        self.model_mock.return_value.alignment = Matrix44.identity()
+        self.model_mock.return_value.instrument_controlled.emit(CommandID.MovePositioner)
+        for i in range(3):
+            self.assertEqual(float(i), float(self.dialog.ftable_widget.item(0, i).text()))
+
+        # Move positioning stack and test only correct fiducial coordinates moved
+        self.mock_instrument.positioning_stack.fkine([0, 10], set_point=True)
+        self.model_mock.return_value.instrument_controlled.emit(CommandID.ChangePositionerBase)
+
+        self.assertEqual(0.0, float(self.dialog.ftable_widget.item(0, 0).text()))
```

Comment:
You only check the first row in all the tests even though there should be 3 rows of data